test/html5/test_tree_construction.rb

```diff
@@ -161,7 +166,7 @@ module Html5libTestCaseParser
   class BadHtml5libFormat < RuntimeError; end
 
   def self.parse_test(test_data)
-    test = { script: :both }
+    test = { script: :off }
```

Comment:
The `html5lib-tests` docs say

> Then there \*may\* be a line that says "\#script-off" or
> "\#script-on". If a line that says "\#script-off" is present, the
> parser must set the scripting flag to disabled. If a line that says
> "\#script-on" is present, it must set it to enabled. Otherwise, the
> test should be run in both modes.

Should we be running all the tests that are not marked `on` or `off` in both modes?

Reply:
I'd forgotten that the absence of `#script-off`/`#script-on` meant we were supposed to run it in both modes (which does explain the `:both`).

I think we should run the tests in both modes. How do you feel about putting that change off for a short while? I've got some refactoring/clean up to the tree construction tests that I was hoping land and it should be easy to make the change to that.

I've published the clean up branch (which was based on an earlier version of this PR) here https://github.com/stevecheckoway/nokogiri/compare/script-enabled-noscript...stevecheckoway:nokogiri:better-error-reporting The high-level idea is to have just a single class, `TestHtml5TreeConstruction`, and define all of the methods using `it` to give them descriptive names that include both the `.dat` file and the line number of the start of the test data.

The `tests.each_with_index` loop can then define all the relevant tests (script on vs. off, fragment with context node or not, etc.)

Reply:
Or I guess I can add that commit to this PR. Whichever is easiest for you!

Reply:
Yup, happy to deal with `:both` in a separate PR!

Reply:
see https://github.com/sparklemotion/nokogiri/pull/3277